## Introduction

As the field of traditional genomics rapidly expands its sequencing technologies and translational abilities, novel applications of genomic data are starting to arise in addressing disease burden.
Complementing the rapid growth in our understanding of human genetic variation was the emergence of genome-wide association studies (GWAS) in the early 2000s to identify gene variants associated with common human diseases.
Non-candidate-driven in design, these observational studies carry out chip array genotyping across population subsamples to subsequently assay for phenotype signal association via statistical approaches in silico.
Measuring averaged allelic effects across all genomics backgrounds and environmental exposures, GWAS have primarily sought to discern genetic association with phenotypes of interest by studying single nucleotide polymorphisms (SNPs) and other DNA variants across the human genome [@doi:10.1371/journal.pcbi.1002822; @doi:10.1038/nrg1521; @doi:10.1038/nrg1522].

In tandem with the movement towards precision medicine, the post-GWAS era strives to bring relevant population-derived gene variants into individual level metrics actionable in health delivery settings.
While GWAS indeed capture gene variants associated with a phenotype of interest on a population level, translating such results to personalized individual metrics of risk requires aggregating contributions of many gene variants in the form of polygenic risk scores (PRS).
PRS provide an ability to explain inherited risk for disease in an individual by representing a weighted sum aggregate of risk alleles based on measured loci effect contributions derived from GWAS [@doi:10.1001/jama.2019.3893; @doi:10.1038/s41576-018-0018-x]. 
In quantifying the effect of particular combinations of genetic SNP variants towards risk prediction, PRS offers a probabilisitic susceptibility value of an individual to disease.
Such genetic risk estimation scores are central to clinical decision-making, serving to reinforce individual health management in heritable disease detection and early prevention of various adult-onset conditions. 
The utility of PRS scores have been demonstrated in previous studies towards disease risk stratification across leading heritable causes of death in the developed world [@doi:10.1038/nature08185; @doi:10.1161/CIRCULATIONAHA.116.024436; @doi:10.1001/jamaoncol.2016.1025; @doi:10.1136/bmj.j5757].

Because common PRS method assumes a simplified genetic architecture consisting of independent weights, understanding interactive relationships among genes and SNPs that associate with disease outcome remain a challenge.
Existing standard multivariate categorical data analysis approaches fall short in handling such enormous possible genetic interaction combinations with both linear and nonlinear effects.
In this context, more robust and efficient methods towards a polygeneic risk calculation are necessary in capturing the overlap between context-dependent effects of both rare and common alleles on human genetic disorder.
Herein, we use the terminology gene-gene (GxG) interactions to indicate any genetic interaction including ones among SNPs that may fall outside of coding regions.

With respect to better understanding the epistasis across an individual's genome, various statistical models have been designed with the intent of capturing high dimensional GxG interactions.
The Multifactor Dimensionality Reduction (MDR) method is one such nonparametric framework that addresses these challenges and has been extensively applied to detect nonlinear complex GxG interactions associated with individual disease [@doi:10.1086/321276; @doi:10.1007/978-1-4939-2155-3_16]. 
By isolating a specific pool of genetic factors from all polymorphism and cross-valiating prediction scores averaged across identified high risk multi-locus genotypes, the original MDR approach is able to categorize multilocus genotypes into two groups of risk based on a threshold value. 
While created with the primary intention towards GxG interaction detection by reducing dimensionality interactively in inferring genotype encodings, the MDR model has additionally demonstrated applicability as a risk score calculation model in constructing PRS scores [@doi:10.1186/1756-0381-6-1].

Modifications built on top of the MDR framework have been proposed in order to better capture multiple significant epistasis models and potential missed interactions owning to limitations of the original model in the higher dimensions.
Model-Based Multifactor Dimensionality Reduction (MB-MDR) was formulated as a flexible GxG detection framework for both dichotomous and continuous traits [@doi:10.1038/ejhg.2011.17; @doi:10.1111/j.1469-1809.2010.00604.x].
Rather than a direct comparison against a threshold level in the original MDR method, MB-MDR merges multilocus genotypes exhibiting significant High or Low risk levels through association testing and adds an additional 'No evidence of risk' categorization. 
In comparison to the standard MDR framework which reveals at most one optimal epistasis model, the MB-MDR method flexibly weighs multiple models by producing a model list ranked with respect to their statistical parameters.

In the present work, we aim to reformulate the PRS leveraging the MB-MDR approach to better capture alternative encodings and epistatic interactions of individual disease risk in a novel Multilocus Risk Score (MRS).
Through the following sections, we briefly review the features of the MDR and MB-MDR software, describe how our new MRS method evaluates polygenic risk, and compare MRS profiling performance to the standard PRS method on evidence-based simulated dataset collections.
In observing prediction accuracy results, we demonstrate the improved performance of our multi-model weighted epistasis framework with inferred genotype encodings over existing PRS methods, showing great potential for more accurate identification of high risk individuals for a specific complex disease.