## Methods

### Multifactor Dimensionality Reduction (MDR) and model-based MDR (MB-MDR)
MDR is a nonparametric method that detects multiple genetic loci associated with a clinical outcome by reducing the dimension of a genotype dataset through pooling multilocus genotypes into high-risk and low-risk groups [@doi:10.1086/321276].
MDR has been applied to a number of real-world datasets and sufficiently identified important variant interactions that associated with various diseases [@doi:10.1186/1479-7364-2-5-318].
Extended from the original MDR algorithm, MB-MDR was first introduced in 2009 [@doi:10.1111/j.1469-1809.2010.00604.x], and its current implementation efficiently and effectively detects multiple sets of significant gene-gene interactions in relation to a trait of interest while efficiently controlling type I error rates via a cross-validation strategy.
By merging multi-locus genotypes exhibiting significant high or low risk based on association testing rather than comparing to an arbitrary threshold as in MDR, MB-MDR provides a flexible framework to detect and measure epistasis. 

In addition to the test statistic and P values associated with each genotype combination, another important output of MB-MDR is the HLO matrices.
Briefly, in the case of a binary trait, for each genotype combination, an HLO matrix is a 3 x 3 matrix with each cell containing H (high), L (low) or O (no evidence), indicating risk of an individual whose genotype pairs fall into that cell [@doi:10.1186/1471-2105-14-138].
For an example binary outcome problem, a genotype combination $SNP_1$ and $SNP_2$ will have an HLO matrix that looks like
$$ \begin{array}{l|ccc}
& SNP_1 = 0 & SNP_1 = 1 & SNP_1 = 2   \\
\hline
SNP_2 = 0 & O        & O        & O \\
SNP_2 = 1 & O        & H        & L \\
SNP_2 = 2 & O        & L        & H
\end{array}
$$
We discuss in the following subsection how these values were utilized in the formulation of the Multilocus Risk Score (MRS).

### From Polygenic Risk Scores (PRS) to Multilocus Risk Scores (MRS)
In this subsection, we quickly review the standard PRS formula then present our modification to this popular risk score calculation.
For both methods, we consider a dataset of $n$ individuals with genomes of $m$ possible SNPs.

In PRS, for each SNP $j$ of an individual $i$, the PRS score is calculated via a summation across $k$ selected SNPs as
$$PRS(i)=\sum_{j=1}^{k} \beta_j \times SNP_{ij}$$
where $\beta_j$ is the weighted risk contribution of the $j^\textrm{th}$ SNP derived from the association test parameters and $SNP_{ij}$ represents the number of minor alleles (0, 1, or 2) at the $j^\textrm{th}$ locus of individual $i$.
Various approaches towards predicting risk of the same disease exist across PRS studies based on the above equation; models may vary according to the specific statistical model used to produce the weights $\beta_j$ for individual genetic variations, the number of genetic variants considered $k$, and the ability of the PRS to generalize to the entire population [@doi:10.1001/jama.2019.3893].

In the MRS framework, we let $k_d$ denote the number of significant combinations for a specific model dimension $d$ (e.g. $d = 2$ results in pairs of SNPs).
In this study, no significance threshold is imposed at the SNP combination level and, thus, $k_d$ reaches its maximum value of $C^d_m$ ($m$ choose $d$).
For each subject $i$ ($i = 1,2, \dotsm, n$), the $d$-way multilocus risk score is calculated as
$$MRS_d(i) = \sum_{j = 1}^{k_d} \gamma_j \times \textrm{HLO}_j(X_{ij})$$
where $\gamma_j$ is the test statistic of the $j^\textrm{th}$ genotype combination output from MB-MDR, $X_{ij}$ is the $j^\textrm{th}$ genotype combinations of subject $i$ and $\textrm{HLO}_j$ represents the $j^\textrm{th}$ recoded HLO matrix (1 = High, -1 = Low, 0 = No evidence).
As an example, consider a pair $X_{*j} = (SNP_{j_1}, SNP_{j_2})$ with $\gamma_j=8.3$ and corresponding HLO matrix of all O's except an L in the first cell.
Then, the contribution of this pair to a subject's risk would be 0 for all subjects except those with genotype 0 at both SNPs.
For the latter, the contribution would be -8.3.

In this study, we consider 1-way and 2-way interactions.
We denote by MRS the combined risk score MRS1 + MRS2.
The significance level of each combination of SNPs on a given dataset is obtained by applying on that dataset the MB-MDR software [@doi:10.1186/1471-2105-14-138; @doi:10.1111/j.1469-1809.2010.00604.x] v.4.4.1.
We will compare the performance of the standard PRS method to the combined risk MRS and also its components, MRS1 and MRS2, separately.

### Mutual information and information gain
For a given simulated data set, we apply entropy-based methods to measure how much information about the phenotype is due to either marginal effects or the synergistic effects of the variants after subtracting the marginal effects.
A dataset's amount of main effect $ME$ can be measured as the total of mutual information between each $SNP_j$ and the phenotypic class $Y$ based on Shannon's entropy $H$ [@doi:10.1002/j.1538-7305.1948.tb01338.x]:
$$ME = \sum_{j} I(SNP_j; Y) = \sum_{j} \left(H(Y) - H(Y|SNP_j)\right).$$

We measure the 2-way interaction information (i.e. degree of synergistic effects of genotypes on the phenotype) of each dataset by summing the pairwise information gain between all pairs of genetic attributes.
Specifically, if we let $X_j$ denote the $j^\textrm{th}$ pair $(SNP_{j_1}, SNP_{j_2})$, the total 2-way interaction gain (i.e. synergistic effects $SE$) is calculated as 
$$SE = \sum_{j} IG(X_j; Y) = \sum_{j} \left(I(SNP_{j_1}, SNP_{j_2}; Y) - I(SNP_{j_1}; Y) - I(SNP_{j_2}; Y)\right),$$
where $IG$ measures how much of the phenotypic class $Y$ can be explained by the 2-way epistatic interaction within the genotype combination $X_j$.
We refer the reader to Ref. [@doi:10.1007/978-1-4939-2155-3_13] for more details on the calculation of the entropy-based terms.

To prevent potential bias, we compute these values from the training set.
However, because the training and holdout sets were randomly split, the amount of main or interaction effect in both datasets are expected to be similar.


### Simulated data
The primary objective of this data simulation process was to provide a comprehensive set of reproducible and diverse datasets for the current study.
Each dataset was generated in the following manner.
For an individual, each genotype was randomly assigned with 1/2 probability of being heterozygous (*Aa*, coded as `1`), 1/4 probability of being homozygous major (*AA*, coded as `0`) and 1/4 probability of being homozygous minor (*aa*, coded as `2`).
The binary endpoint for the data was determined using a recently proposed evolutionary-based method for dataset generation called Heuristic Identification of Biological Architectures for simulating Complex Hierarchical Interactions [@doi:10.1142/9789813235533_0024].
This method uses genetic programming to build different mathematical and logical models resulting in a binary endpoint, such that the objective function called fitness is maximized.
In this study, to arrive at a diverse collection of datasets, we aim to maximize the difference in predictive performance of all pairs of ten pre-selected classifiers.
Details on data simulation are provided in the README of the study's analysis repository [https://github.com/lelaboratoire/rethink-prs/](https://github.com/lelaboratoire/rethink-prs/).

The final collection has 450 datasets containing 1000 individuals and 10 SNPs with various amount of epistatic effect on the binary phenotypic outcome.
For each simulated dataset, after randomly splitting the entire data in two smaller sets (80% training and 20% holdout), we built the MRS model on training data to obtain the $\gamma$ coefficients and the HLO matrix, and then we calculated risk score for each individual in the holdout set.
We assess the performance of the MRS by comparing the area under the Receiving Operator Characteristic curve (auROC) with that of the standard PRS method on the holdout set.

### Manuscript drafting
This manuscript is collaboratively written using Manubot, a software for writing scholarly documents via GitHub [@doi:10.1371/journal.pcbi.1007128].
With continuous integration, Manubot automatically updates the manuscript when its authors approve the changes.
As a result, the latest version of this manuscript is always available for review at [https://lelaboratoire.github.io/rethink-prs-ms/](https://lelaboratoire.github.io/rethink-prs-ms/).


### Availability
Detailed simulation and analysis code needed to reproduce the results in this study is available at [https://github.com/lelaboratoire/rethink-prs/](https://github.com/lelaboratoire/rethink-prs/).

